src/Entities/Player.cpp

```diff
@@ -209,6 +213,15 @@ void cPlayer::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 		}
 	}
 
+	Vector3d CurrPos = GetPosition();
+	std::chrono::steady_clock::duration timeDiff = std::chrono::steady_clock::now() - m_LastTickTime;
+	double timeRatio = 1 / (double(timeDiff.count()) * std::chrono::steady_clock::period::num / std::chrono::steady_clock::period::den);
+
+	SetSpeedValue(new Vector3d((m_LastPosition.x - CurrPos.x)*timeRatio, (m_LastPosition.y - CurrPos.y)*timeRatio, (m_LastPosition.z - CurrPos.z)*timeRatio));
```

Comment:
This is a memory leak. The new is constructing a pointer which is then copy constructed into a value type. @madmaxoft Is there any reason we can't delete the `Vector3(const Vector3<_T> * a_Rhs)` constructor? It is pretty dangerous.

Also this code looks generally weird. The Vector3 family of types provides arithmetic methods. Is there any reason to do this rather than `(m_LastPosition - CurrPos) * timeRatio`?